src/files/files.service.ts

```diff
@@ -1,79 +1,90 @@
-import { Injectable, NotFoundException } from '@nestjs/common';
+import { BadRequestException, Inject, Injectable, NotFoundException } from '@nestjs/common';
 import { InjectRepository } from '@nestjs/typeorm';
-import { S3 } from 'aws-sdk';
 import { Repository } from 'typeorm';
+import { ClientProxy } from '@nestjs/microservices';
+import axios from 'axios';
+import { v4 as uuid } from 'uuid';
 import { FilesEntity } from './files.entity';
 import { ApiConfigService } from '../../common/config/api-config.service';
-import { v4 as uuid } from 'uuid';
+import { FILE_SERVICE } from '../../common/constants/rmq-constants';
 import { FileDto } from './dto/file.dto';
-import { createReadStream } from 'fs';
+import { firstValueFrom } from 'rxjs';
 
 @Injectable()
 export class FilesService {
-    bucketName = this.configService.AWS.bucketName;
     constructor(
         @InjectRepository(FilesEntity)
         private filesRepo: Repository<FilesEntity>,
         private readonly configService: ApiConfigService,
+        @Inject(FILE_SERVICE) private readonly fileClient: ClientProxy,
     ) {}
 
     async createFile(createFileDto: FileDto, postId: number): Promise<FilesEntity> {
-        const newFile = await this.filesRepo.create({
-            key: createFileDto.key,
-            url: createFileDto.url,
-            post: { id: postId },
-        });
-        return this.filesRepo.save(newFile);
+        try {
+            const newFile = await this.filesRepo.create({
+                key: createFileDto.key,
+                url: createFileDto.url,
+                post: { id: postId },
+            });
+            return this.filesRepo.save(newFile);
```

Comment:
`try/catch` does not catch the error if `this.filesRepo.save` is async function and `await` is not used.
Please update it as follows:
`return await this.filesRepo.save(newFile);`